vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java

```diff
@@ -36,37 +38,82 @@ public abstract class LinkManager<L extends ILink<T>, T extends IVSpaceElement,
 
     @Autowired
     private IStorageEngine storage;
+    
+    @Autowired
+    private IImageService imageService;
+
+    protected abstract void deleteLinkRepo(L link);
+
+    protected abstract void deleteLinkDisplayRepo(L link);
+
+    protected abstract void removeFromLinkList(ISpace space, L link);
+
+    protected abstract U updateLinkAndDisplay(L link, U displayLink);
+
+    protected abstract U getDisplayLink(String linkDisplayId) throws LinkDoesNotExistsException;
+
+    protected abstract L getLink(String linkId);
+
+    protected abstract L createLinkObject(String title, String id) throws SpaceDoesNotExistException;
 
+    protected abstract T getTarget(String linkedId);
+
+    protected abstract U createDisplayLink(L link);
+    
     @Override
     public U createLink(String title, String id, float positionX, float positionY, int rotation, String linkedId,
-            String linkLabel, DisplayType displayType, byte[] linkImage, String imageFilename)
-            throws SpaceDoesNotExistException, ImageCouldNotBeStoredException, SpaceDoesNotExistException {
+            String linkLabel, String linkDesc, DisplayType displayType, byte[] linkImage, String imageFilename, String existingImageId)
+            throws SpaceDoesNotExistException, ImageCouldNotBeStoredException, ImageDoesNotExistException {
 
+        if (title == null || title.trim().isEmpty()) {
+            throw new IllegalArgumentException("Title cannot be null or empty.");
+        }
+        if (id == null || id.trim().isEmpty()) {
+            throw new IllegalArgumentException("ID cannot be null or empty.");
+        }
+        if (linkedId == null || linkedId.trim().isEmpty()) {
+            throw new SpaceDoesNotExistException("Linked ID cannot be null or empty.");
+        }
+        if (displayType == null) {
+            throw new IllegalArgumentException("DisplayType cannot be null.");
+        }
+        if ((linkImage == null || linkImage.length == 0) && (existingImageId == null || existingImageId.trim().isEmpty())) {
+            throw new ImageDoesNotExistException("Either linkImage or existingImageId must be provided.");
```

Comment:
wrong exception to throw. Image does not exist is different than no image provided.